Fix issue #59 - Update contributing.md doc (#63)
trypear/pearai-app: CONTRIBUTING.md

The extension can be run in two ways:

RECOMMENDED: Internally within the main PearAI application (which is a VSCode fork): <https://github.com/trypear/pearai/>. This guide is for running it internally.

> Standalone as an extension. For running it standalone, you will want to `cd` into `extensions/pearai-extension` and visit [Contributing to pearai-extension](extensions/pearai-extension/CONTRIBUTING.md)

Reviewer: ```suggestion
```

> NOTE: If you need to debug the 32bit version of VS Code on 64bit Windows, follow [the guide on how to do that](https://github.com/microsoft/vscode/wiki/Build-and-run-32bit-Code---OSS-on-Windows).

### Getting the sources

First, fork the `trypear/pearai-app` repository so that you can make pull requests. Then, clone your fork locally:

```
git clone https://github.com/<<<your-github-account>>>/pearai-app.git
```

Occasionally you will want to merge changes in the upstream repository (the official code repo) with your fork.

```
cd pearai-app
git checkout main
git pull https://github.com/trypear/pearai-app.git main
```

Manage any merge conflicts, commit them, and then push them to your fork.

**Note**: The `trypear/pearai-app` repository contains a collection of GitHub Actions that help us with triaging issues. They are disabled by default, but you can enable them for your fork via `https://github.com/<<Your Username>>/pearai-app/actions`.

### Build

Install and build all of the dependencies using `Yarn`:

```
cd pearai-app
yarn
```

Then you have two options:

> - If you want to build from inside VS Code, you can open the `pearai-app` folder and start the build task with <kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>B</kbd> (<kbd>CMD</kbd>+<kbd>Shift</kbd>+<kbd>B</kbd> on macOS). The build task will stay running in the background even if you close the app.

Reviewer: ```suggestion
- If you want to build from inside VS Code, you can open the `pearai-app` folder and start the build task with <kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>B</kbd> (<kbd>CMD</kbd>+<kbd>Shift</kbd>+<kbd>B</kbd> on macOS). The build task will stay running in the background even if you close the app. If you happen to close VS Code and open it again, just resume the build by pressing <kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>B</kbd> (<kbd>CMD</kbd>+<kbd>Shift</kbd>+<kbd>B</kbd>) again. You can kill it by running the `Kill Build VS Code` task or pressing <kbd>Ctrl</kbd>+<kbd>D</kbd> in the task terminal.
```

.\scripts\code-server.bat --launch
```

> You can identify the development version of the PearAI App ("Code - OSS") by the following icon in the Dock or Taskbar:

Reviewer: ```suggestion
You can identify the development version of the PearAI App by finding the Pear logo in your Dock or Taskbar.
```

> [![PearAI App default icon](https://i.imgur.com/D2CeX0y.png)](https://i.imgur.com/D2CeX0y.png)

Reviewer: ```suggestion
```

## Work Branches

Even if you have push rights on the trypear/pearai-app repository, you should create a personal fork and create feature branches there when you need them. This keeps the main repository clean and your personal workflow cruft out of sight.

## Pull Requests

Before we can accept a pull request from you, you'll need to sign a [[Contributor License Agreement (CLA)|Contributor-License-Agreement]]. It is an automated process and you only need to do it once.

To enable us to quickly review and accept your pull requests, always create one pull request per issue and [link the issue in the pull request](https://github.com/blog/957-introducing-issue-mentions). Never merge multiple requests in one unless they have the same root cause. Be sure to follow our [[Coding Guidelines|Coding-Guidelines]] and keep code changes as small as possible. Avoid pure formatting changes to code that has not been modified otherwise. Pull requests should contain tests whenever possible.

### Introducing usage of new Electron API with a PR

> A pull request that depends on Electron API that VS Code is currently not using comes with a certain risk and may be rejected. Whenever we update Electron, there is a chance that less popular Electron APIs break and it is very hard to find out upfront. Once a PR lands in VS Code, the role of maintaining the feature moves to the team and as such we have to follow up with upstream components to ensure the feature is still supported. As such, as a rule of thumb:

Author: This section probably needs to be reworded.
